plugins/ginkgo/src/ginkgo_linear_algebra.cpp

```diff
@@ -0,0 +1,161 @@
+
+  void axpy(Arccore::Real alpha, const Vector& x,
+            Vector& r) const override;
+
+  void aypx(Arccore::Real alpha, Vector& y,
+            const Vector& x) const override;
+
+  void copy(const Vector& x, Vector& r) const override;
+
+  Arccore::Real dot(const Vector& x, const Vector& y) const override;
+
+  void scal(Arccore::Real alpha, Vector& x) const override;
+
+  void diagonal(const Matrix& a, Vector& x) const override;
+
+  void reciprocal(Vector& x) const override;
+
+  void pointwiseMult(const Vector& x, const Vector& y,
+                     Vector& w) const override;
+};
+
+Arccore::Real
+InternalLinearAlgebra::norm0(const Vector& vx ALIEN_UNUSED_PARAM) const
+{
+  throw Arccore::NotImplementedException(A_FUNCINFO, "GinkgoLinearAlgebra::norm0 not implemented");
+}
+
+Arccore::Real
+InternalLinearAlgebra::norm1(const Vector& vx) const
+{
+  throw Arccore::NotImplementedException(A_FUNCINFO, "GinkgoLinearAlgebra::norm1 not implemented");
+}
+
+Arccore::Real
+InternalLinearAlgebra::norm2(const Vector& vx) const
+{
+  using vec = gko::matrix::Dense<double>;
+  auto exec = vx.internal()->get_executor();
+  auto mtx_res = gko::initialize<vec>({ 1.0 }, exec);
+  vx.internal()->compute_norm2(gko::lend(mtx_res.get()));
+  return mtx_res->get_values()[0];
+}
+
+void InternalLinearAlgebra::mult(const Matrix& ma, const Vector& vx, Vector& vr) const
+{
+  ma.internal()->apply(lend(vx.internal()), lend(vr.internal()));
+}
+
+void InternalLinearAlgebra::axpy(
+Arccore::Real alpha, const Vector& vx, Vector& vr) const
+{
+  using vec = gko::matrix::Dense<double>;
+  auto exec = vx.internal()->get_executor();
+  auto mtx_alpha = gko::initialize<vec>({ alpha }, exec);
+  vr.internal()->add_scaled(mtx_alpha.get(), vx.internal());
+}
+
+void InternalLinearAlgebra::copy(const Vector& vx, Vector& vr) const
+{
+  vr.internal()->copy_from(vx.internal());
+}
+
+Arccore::Real
+InternalLinearAlgebra::dot(const Vector& vx, const Vector& vy) const
+{
+  //~ PetscScalar dot_prod;
+  //~ VecDot(vx.internal(), vy.internal(), &dot_prod);
+  //~ return static_cast<Arccore::Real>(dot_prod);
+  return {};
+}
+
+void InternalLinearAlgebra::diagonal(Matrix const& m ALIEN_UNUSED_PARAM, Vector& v ALIEN_UNUSED_PARAM) const
+{
+  //~ MatGetDiagonal(m.internal(), v.internal());
+  // return{};
+}
+
+void InternalLinearAlgebra::reciprocal(Vector& v ALIEN_UNUSED_PARAM) const
+{
+  //~ VecReciprocal(v.internal());
+}
+
+void InternalLinearAlgebra::aypx(
+Arccore::Real alpha, Vector& y, const Vector& x) const
+{
+  //~ VecAYPX(y.internal(), alpha, x.internal()); // y = x + alpha y
+}
+
+void InternalLinearAlgebra::pointwiseMult(
+const Vector& x, const Vector& y, Vector& w) const
+{
+  //VecPointwiseMult(
+  //~ w.internal(), x.internal(),
+  //~ y.internal()); // Computes the componentwise multiplication w = x*y.
+}
+
+void InternalLinearAlgebra::scal(Arccore::Real alpha, Vector& x) const
+{
+  //   VecScale(x.internal(), alpha);
+}
```

Comment:
Not available in Ginkgo ?
At least should return a warning as not implemented.

Reply:
Wasn't fihished ! Updated now :)